packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -304,9 +297,67 @@ export class OdspDocumentService implements IDocumentService {
         });
     }
 
+    private clearJoinSessionTimer() {
+        if (this.joinSessionRefreshTimer !== undefined) {
+            clearTimeout(this.joinSessionRefreshTimer);
+            this.joinSessionRefreshTimer = undefined;
+        }
+    }
+
+    private async refreshSessionPeriodically(requestWebsocketTokenFromJoinSession: boolean) {
+        for (let i = 0;;i++) {
```

Comment:
if compiler allows it, prefer for(;;) or while(true) (if needed - you can disable eslint rule)